packages/purgecss-webpack-plugin/src/index.ts

```diff
@@ -127,7 +128,17 @@ export default class PurgeCSSPlugin {
 
         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
         // @ts-ignore
-        compilation.updateAsset(name, new ConcatSource(purged.css));
+        let source: Source = new ConcatSource(purged.css)
```

Comment:
Any reason why you are re-using the `source` variable? I would suggest something more in the original style like:

```ts
compilation.updateAsset(name, new ConcatSource(purged.css));
if (purged.rejectedCss !== undefined) {
  const rejectedName = path.dirname(name) + '/' + path.basename(name, '.css') + '-rejected' + path.extname(name);
  const source = new ConcatSource(purged.rejectedCss);
  if (compilation.getAsset(rejectedName) {
    compilation.emitAsset(rejectedName, source);  
  } else {
   compilation.updateAsset(rejectedName, source);
  }
}
```

Some minor nitpick would be to maintain consistent with use of semi-colons in the file.